src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -20857,6 +20932,69 @@ static void F(C? x, C y)
                 Diagnostic(ErrorCode.WRN_NullReferenceArgument, "x").WithArguments("c", "C.implicit operator B(C c)").WithLocation(12, 22));
         }
 
+        [Fact]
+        public void Tuple_OtherMembers()
+        {
+            var source =
+@"internal delegate T D<T>();
+namespace System
+{
+    public struct ValueTuple<T1, T2>
+    {
+        public ValueTuple(T1 item1, T2 item2)
+        {
+            Item1 = item1;
+            Item2 = item2;
+            E2 = null;
+        }
+        public T1 Item1;
+        public T2 Item2;
+        internal T1 P1 => Item1;
+        internal event D<T2>? E2;
+    }
+}
+class C
+{
+    static void F(object? x)
+    {
+        var y = (x, x);
+        y.P1.ToString();
+        y.E2?.Invoke().ToString();
+        if (x == null) return;
+        var z = (x, x);
+        z.P1.ToString();
+        z.E2?.Invoke().ToString();
+    }
+}";
+            // PROTOTYPE(NullableReferenceTypes): Should not report warnings for `z.P1` or `z.E2?.Invoke()`.
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (24,11): error CS0070: The event '(object?, object?).E2' can only appear on the left hand side of += or -= (except when used from within the type '(object?, object?)')
+                //         y.E2?.Invoke().ToString();
+                Diagnostic(ErrorCode.ERR_BadEventUsage, "E2").WithArguments("(object?, object?).E2", "(object?, object?)").WithLocation(24, 11),
+                // (28,11): error CS0070: The event '(object?, object?).E2' can only appear on the left hand side of += or -= (except when used from within the type '(object?, object?)')
+                //         z.E2?.Invoke().ToString();
+                Diagnostic(ErrorCode.ERR_BadEventUsage, "E2").WithArguments("(object?, object?).E2", "(object?, object?)").WithLocation(28, 11),
+                // (23,9): warning CS8602: Possible dereference of a null reference.
+                //         y.P1.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "y.P1").WithLocation(23, 9),
+                // (24,9): hidden CS8607: Expression is probably never null.
```

Comment:
>/ (24,9): hidden CS8607: Expression is probably never null. [](start = 17, length = 59)

Is this accurate?

Reply:
Good catch. `BoundBadExpression` was being treated as non-nullable.

---
In reply to: [190753053](https://github.com/dotnet/roslyn/pull/26356#discussion_r190753053) [](ancestors = 190753053)